src/utils/AuthorizationPersistance.js

```diff
@@ -0,0 +1,89 @@
+const db = require('../storage/DataAccess');
+
+const SERVERS = 'serverConfig';
+const CLIENTS = 'clientConfig';
```

Comment:
These are two different collections right? I think we only need one (which is called `servers` right now). The app doesn't need to know the `jwks_url` of the clients registered with it. This is all handled via Keycloak. What the app does need to know is the servers it has registered as a client with. 

For example:
EHR registers with Backend App via Keycloak. The app itself does not need to know anything about this. The Backend App then registers with the EHR (shares its own jwks_url which is just `{backend app base url}/jwks`). The EHR will then return a client ID. The Backend app will then use this when creating the JWT client_assertion and obtain an access token. So the new data to store in the `servers` collection is `clientId` and `token`. 

Note there is already a [schema for the collection on the wiki](https://github.com/mcode/medmorph-backend/wiki/Collections) (certainly open to new ideas on now to present this since we're not using TypeScript). So the new schema would be

```js
{
   name: String // human readable name
   endpoint: String // FHIR base url
   type: 'KA' | 'EHR' | 'PHA' // type of server the endpoint is
   clientId: String // the client ID assigned to this app during registration
   token: String // the last access token received from authorization
   tokenExp: number // the time the access token expires
}
```

With all of that said my comments for change are:
- [ ] replace this with a more generic `servers` data access layer (i.e. `getServerClientId(serverId)`)
- [ ] link this to your `Client` class to get the `clientId` for `generateJWT` method
- [ ] update `connectToServer` in `Client` class to read/write the `token` and `tokenExp`

Note: since these recommendations are substantial changes I believe we agreed to get a second opinion before implementing them.